src/twisted/internet/base.py

```diff
@@ -567,7 +739,15 @@ def resolve(self, name, timeout = (1, 3, 11, 45)):
             return defer.succeed('0.0.0.0')
         if abstract.isIPAddress(name):
             return defer.succeed(name)
-        return self.resolver.getHostByName(name, timeout)
+        d = self.resolver.getAddressInformation(name, 0)
+        def cbGotInfo(addresses):
+            for info in addresses:
+                if info.family == socket.AF_INET:
+                    return info.address[0]
+            # XXX Test me
```

Comment:
This code is in fact covered by tests so I'm not sure what this comment is referring to; it should be removed.